File: src/main/java/com/sica/backend/repository/JustificationRepository.java
package com.sica.backend.repository;

import com.sica.backend.entity.Justification;
import com.sica.backend.entity.JustificationStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface JustificationRepository extends JpaRepository<Justification, String> {
    List<Justification> findByStudentPersonId(String studentId);
    List<Justification> findByStatus(JustificationStatus status);
    Justification findByAttendanceId(String attendanceId);

    @Query("SELECT j FROM Justification j WHERE j.professor.id = :professorId")
    List<Justification> findByProfessorId(String professorId);

    @Query("SELECT j FROM Justification j WHERE j.professor.id = :professorId AND j.status = :status")
    List<Justification> findByProfessorIdAndStatus(String professorId, JustificationStatus status);
}

File: src/main/kotlin/com/sica/backend/repository/JustificationRepository.kt
package com.sica.backend.repository

import com.sica.backend.entity.Justification
import com.sica.backend.entity.JustificationStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface JustificationRepository : JpaRepository<Justification, String> {
    fun findByStudentPersonId(studentId: String): List<Justification>
    fun findByStatus(status: JustificationStatus): List<Justification>
    fun findByAttendanceId(attendanceId: String): Justification?
    
    @Query("SELECT j FROM Justification j WHERE j.professor.id = :professorId")
    fun findByProfessorId(professorId: String): List<Justification>
    
    @Query("SELECT j FROM Justification j WHERE j.professor.id = :professorId AND j.status = :status")
    fun findByProfessorIdAndStatus(professorId: String, status: JustificationStatus): List<Justification>
}
